@navikt/core/react/src/form/combobox/FilteredOptions/filteredOptionsContext.tsx

```diff
@@ -151,102 +152,38 @@ export const FilteredOptionsProvider = ({ children, value: props }) => {
-  }, []);
-
-  useEffect(() => {
-    if (filteredOptionsIndex !== null && isListOpen) {
-      scrollToOption(filteredOptionsIndex);
-    }
-  }, [filteredOptionsIndex, isListOpen, scrollToOption]);
-
-  const moveFocusToInput = useCallback(() => {
-    setFilteredOptionsIndex(null);
-    toggleIsListOpen(false);
-  }, [toggleIsListOpen]);
-
-  const moveFocusToEnd = useCallback(() => {
-    const lastIndex = filteredOptions.length - 1;
-    toggleIsListOpen(true);
-    setFilteredOptionsIndex(lastIndex);
-  }, [filteredOptions.length, toggleIsListOpen]);
+  // TODO: Re-write or remove after re-write?
+  const currentOption = useMemo(
+    () =>
+      filteredOptionsMap[virtualFocus.activeElement?.getAttribute("id") || -1],
+    [filteredOptionsMap, virtualFocus]
+  );
 
+  // TODO: Can be deleted if we move toggleIsListOpen(false) to the event handling in Input.tsx
   const moveFocusUp = useCallback(() => {
-    if (filteredOptionsIndex === null) {
-      return;
-    }
-    if (filteredOptionsIndex === getMinimumIndex()) {
+    if (virtualFocus.isFocusOnTheTop) {
       toggleIsListOpen(false);
-      setFilteredOptionsIndex(null);
-    } else {
-      const newIndex = Math.max(getMinimumIndex(), filteredOptionsIndex - 1);
-      setFilteredOptionsIndex(newIndex);
     }
-  }, [filteredOptionsIndex, getMinimumIndex, toggleIsListOpen]);
+    virtualFocus.moveFocusUp();
+  }, [toggleIsListOpen, virtualFocus]);
 
+  // TODO: Can be deleted if we move toggleIsListOpen(true) to the event handling in Input.tsx
   const moveFocusDown = useCallback(() => {
-    if (filteredOptionsIndex === null || !isListOpen) {
+    if (virtualFocus.activeElement === null || !isListOpen) {
       toggleIsListOpen(true);
-      if (allowNewValues || filteredOptions.length >= 1) {
-        setFilteredOptionsIndex(getMinimumIndex());
-      }
-      return;
     }
-    const newIndex = Math.min(
-      filteredOptionsIndex + 1,
-      Math.max(getMinimumIndex(), filteredOptions.length - 1)
-    );
-    setFilteredOptionsIndex(newIndex);
-  }, [
-    allowNewValues,
-    filteredOptions.length,
-    filteredOptionsIndex,
-    getMinimumIndex,
-    isListOpen,
-    toggleIsListOpen,
-  ]);
+    virtualFocus.moveFocusDown();
+  }, [isListOpen, toggleIsListOpen, virtualFocus]);
 
-  const activeDecendantId = useMemo(() => {
-    if (filteredOptionsIndex === null) {
-      return undefined;
-    } else if (filteredOptionsIndex === -1) {
-      return `${id}-combobox-new-option`;
-    } else {
-      return `${id}-option-${currentOption?.replace(" ", "-")}`;
-    }
-  }, [filteredOptionsIndex, currentOption, id]);
+  const activeDecendantId = useMemo(
+    () => virtualFocus.activeElement?.getAttribute("id") || undefined,
+    [virtualFocus.activeElement]
+  );
 
   const filteredOptionsState = {
     activeDecendantId,
     allowNewValues,
-    filteredOptionsRef,
-    filteredOptionsIndex,
-    setFilteredOptionsIndex,
+    filteredOptionsRef: setFilteredOptionsRef,
```

Comment:
Hvorfor ikke bruke samme navn?

Reply:
Dette skyldes endring fra å bruke useRef til useState for å holde på filteredOptionsRef, som foreslått av @KenAJoh. Husker ikke helt argumentasjonen for det, men mener det gikk på at vi ønsket at endringer i elementet skulle føre til at useVirtualFocus re-intialiserte seg (tror jeg?).

Da sitter vi med både filteredOptionsRef og setFilteredOptionsRef, men utenfor denne filen vil jeg gjerne fremdeles se på det som en ref, og dermed renamet jeg ikke den.

Men er åpen for å bli overbevist om det motsatte.

Reply:
Tror ikke ref-en blir satt på nytt ved endringer i elementet, bare når det mountes og unmountes.

Reply:
Jeg kan nok ha misforstått noe her. Jeg skal sjekke med Ken hva som var greien her.

Reply:
Vil være risiko for at `useVirtualFocus` ikke blir oppdatert riktig hvis `filteredOptionsRef` er en `React.useRef` da den ikke fører til state-updates ved endring. 
```
const virtualFocus = useVirtualFocus(filteredOptionsRef);
```
`filteredOptionsRef` vil her potensielt være `null` ved første kjøring, men vil kjøres på nytt når `setFilteredOptionsRef(ref)` blir satt.

I praksis blir sikkert `useVirtualFocus` kjørt på nytt uansett da andre state-updates oppdaterer `useVirtualFocus`, så er bare initial mount som blir påvirket. 

(Har ikke testet om forskjellen påvirker faktisk kode da)

Reply:
Kan kjøpe den, spesielt siden useVirtualFocus ikke ligger i en useEffect.